numpy/random/generator.pyx

```diff
@@ -434,6 +434,8 @@ cdef class Generator:
 
         """
         if high is None:
+            if isinstance(low, (int, np.integer)) and low <= 0:
```

Comment:
I'm not really a fan of this type of detection - it doesn't work on things that are coercible to integers.

Could you instead use something like the following?
```python
high_and_low_swapped = high is None:
if high_and_low_swapped:
    high = low
    low = 0

try:
    # the chain of ifs below
except ValueError as e:
    if 'low >= high' == str(e):
        raise ValueError(...) from None
```

Reply:
Thanks @eric-wieser . Just pushed the changes.